### PR TITLE
Add style check to code check and run cargo fmt

### PR DESCRIPTION
**Pull request by EthanPlant**
Modifies `codecheck.py` to add code style checking to CI.

Also runs `cargo fmt` on the project to clean up code styling

### CODE REVIEW

File: codecheck.py [hunk 1]
```diff
@@ -27,6 +26,24 @@ def run_grep(lookup, filename) -> str:
...
+                "--check",
+                "--manifest-path",
+                "charlotte_core/Cargo.toml",
+            ],
+            check=True,
+            stderr=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+        )
+    except subprocess.CalledProcessError:
+        print("Style issues detected!")
```
Comment (GranitBebok) [inline codecheck.py]: Maybe add a little note like, `style issues detected please run 'cargo fmt'`

**Review by xadaemon — CHANGES_REQUESTED** [review]
See comment on the code check.py

File: codecheck.py (hunk 1 above)
Comment (EthanPlant) [reply codecheck.py]: Done.

**Review by xadaemon — APPROVED** [review]
👍🏻 lgtm

**Pull request merged**